blocks/columns/columns.css

```diff
@@ -53,6 +53,13 @@
   margin: 0 auto 1rem;
 }
 
+.columns.heading-divider h2,
```

Comment:
Instead of restricting to just the columns block, we can add a global style for underlined headings (like `.underlined.h2` etc.) and then use it in columns and other places as needed,

Reply:
Maybe a section-level selector to add underline to all headings in section.